bindings/python/crocoddyl/multibody/contacts/contact-1d.cpp

```diff
@@ -0,0 +1,122 @@
+          ":param gains: gains of the contact model (default np.matrix([0.,0.]))"))
+      .def(bp::init<boost::shared_ptr<StateMultibody>, FrameTranslation, std::size_t, bp::optional<Eigen::Vector2d> >(
+          bp::args("self", "state", "xref", "nu", "gains"),
+          "Initialize the contact model.\n\n"
+          ":param state: state of the multibody system\n"
+          ":param xref: reference frame translation\n"
+          ":param nu: dimension of control vector\n"
+          ":param gains: gains of the contact model (default np.matrix([ [0.],[0.] ]))"))
+      .def(bp::init<boost::shared_ptr<StateMultibody>, FrameTranslation, bp::optional<Eigen::Vector2d> >(
+          bp::args("self", "state", "xref", "gains"),
+          "Initialize the contact model.\n\n"
+          ":param state: state of the multibody system\n"
+          ":param Mref: reference frame translation\n"
+          ":param gains: gains of the contact model (default np.matrix([ [0.],[0.] ]))"))
+      .def("calc", &ContactModel1D::calc, bp::args("self", "data", "x"),
+           "Compute the 1D contact Jacobian and drift.\n\n"
+           "The rigid contact model throught acceleration-base holonomic constraint\n"
+           "of the contact frame placement.\n"
+           ":param data: contact data\n"
+           ":param x: state point (dim. state.nx)")
+      .def("calcDiff", &ContactModel1D::calcDiff, bp::args("self", "data", "x"),
+           "Compute the derivatives of the 1D contact holonomic constraint.\n\n"
+           "The rigid contact model throught acceleration-base holonomic constraint\n"
+           "of the contact frame placement.\n"
+           "It assumes that calc has been run first.\n"
+           ":param data: cost data\n"
+           ":param x: state point (dim. state.nx)")
+      .def("updateForce", &ContactModel1D::updateForce, bp::args("self", "data", "force"),
+           "Convert the force into a stack of spatial forces.\n\n"
+           ":param data: cost data\n"
+           ":param force: force vector (dimension 1)")
+      .def("createData", &ContactModel1D::createData, bp::with_custodian_and_ward_postcall<0, 2>(),
+           bp::args("self", "data"),
+           "Create the 1D contact data.\n\n"
+           "Each contact model has its own data that needs to be allocated. This function\n"
+           "returns the allocated data for a predefined cost.\n"
+           ":param data: Pinocchio data\n"
+           ":return contact data.")
+      .add_property("reference",
+                    bp::make_function(&ContactModel1D::get_reference, bp::return_value_policy<bp::return_by_value>()),
+                    &ContactModel1D::set_reference, "reference contact translation")
+      .add_property("xref",
+                    bp::make_function(&ContactModel1D::get_xref, deprecated<>("Deprecated. Use id or reference.")),
+                    "reference frame translation")
```

Comment:
Also remove this line.
We don't want to push this deprecated method.
We deprecate a function when it has been previously develop.
This helps users to upgrade users' code.